nebula2/gclient/net/Session.py

```diff
@@ -39,24 +39,97 @@ def execute(self, stmt):
         except IOErrorException as ie:
             if ie.type == IOErrorException.E_CONNECT_BROKEN:
                 self._pool.update_servers_status()
                 if self._retry_connect:
                     if not self._reconnect():
                         logging.warning('Retry connect failed')
-                        raise IOErrorException(IOErrorException.E_ALL_BROKEN, ie.message)
+                        raise IOErrorException(
+                            IOErrorException.E_ALL_BROKEN, ie.message)
                     resp = self._connection.execute(self._session_id, stmt)
                     end_time = time.time()
                     return ResultSet(resp,
-                                     all_latency=int((end_time - start_time) * 1000000),
+                                     all_latency=int(
+                                         (end_time - start_time) * 1000000),
                                      timezone_offset=self._timezone_offset)
             raise
         except Exception:
             raise
 
+    def execute_json(self, stmt):
+        """execute statement and return the result as a JSON string
+            Date and Datetime will be returned in UTC
+            JSON struct:
+              "results": [
+               {
+                  "columns": [],
+                  "data": [
+                    {
+                       "row": [ row-data ],
+                       "meta": [ metadata ]
+                    },
+                  ],
+                  "latencyInUs" : 0,
+                  "spaceName": "",
+                  "planDesc ": {
+                    "planNodeDescs": [ {
+                      "name" : "",
+                      "id" : 0,
+                      "outputVar" : "",
+                      "description" : {"key" : ""},
+                      "profiles" : [{
+                        "rows" : 1,
+                        "execDurationInUs" : 0,
+                        "totalDurationInUs" : 0,
+                        "otherStats" : {}, // map
+                      }],
+                      "branchInfo" : {
+                        "isDoBranch" : false,
+                        "conditionNodeId" : -1,
+                      },
+                      "dependencies" : [] // vector of ints
+                      }
+                    ],
+                    "nodeIndexMap" : {},
+                    "format" : "",
+                    "optimize_time_in_us" : 0,
+                  },
+                  "comment ": "",
+                  "errors" : "" // errorMsg
+                }
+              ]
+            	}]
+
+        :param stmt: the ngql
+        :return: JSON string
+        """
+        if self._connection is None:
+            raise RuntimeError('The session has released')
+        try:
+            start_time = time.time()
+            resp_json = self._connection.execute_json(self._session_id, stmt)
+            end_time = time.time()
```

Comment:
what's the usage about the start_time and end_time?

Reply:
These are used in the `execute` interface. Deleted.